synapse/handlers/cas.py

```diff
@@ -130,7 +130,9 @@ async def _validate_ticket(
         except PartialDownloadError as pde:
             # Twisted raises this error if the connection is closed,
             # even if that's being used old-http style to signal end-of-data
-            body = pde.response
+            # Cast safety: Error.response is Optional[bytes]. but a PartialDownloadError
+            # always has a non-None response.
+            body = cast(bytes, pde.response)
```

Comment:
can we instead `assert body is not None` ?

Reply:
Not quite, but I've done something equivalent in 149fecc88.